### PR TITLE
Data Type mismatch Fix

### PR DESCRIPTION
**Pull request by gucifer**
Description:
Fixed Data Type error in Nvidia Apex.
Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by vfdev-5** [comment]
@gucifer I think you can use `.to(tensor)` to convert to appropriate dtype and do it only in one place instead of two places...

**Comment by sdesrozis** [comment]
@gucifer IMO data should be moved in `_extract_features()`, so it would be done once.

**Comment by gucifer** [comment]
> @gucifer I think you can use `.to(tensor)` to convert to appropriate dtype and do it only in one place instead of two places...
> @gucifer IMO data should be moved in `_extract_features()`, so it would be done once.
I tried that by changing last line of `_extract_features` to `return outputs.to(torch.float64)` but it does not seem to work. Anything, that requires class to class communication even from `_BaseInceptionMetric` to `FID/IS` gives the same error.

**Comment by sdesrozis** [comment]
> Anything, that requires class to class communication even from `_BaseInceptionMetric` to `FID/IS` gives the same error.

What do you mean by `class communication` ?

**Comment by gucifer** [comment]
> > Anything, that requires class to class communication even from `_BaseInceptionMetric` to `FID/IS` gives the same error.
> 
> What do you mean saying "class communication" ? Could you provide a snippet to illustrate ?

```
class A:
    def __init__(self):
        self.a = 0
    def func(self):
        self.a += B().func()

class B:
    def __init__(self):
        self.b = 1
    def func(self):
        return self.b
```
What I mean is parameter passing from method of one class to another.

**Comment by sdesrozis** [comment]
So weird... It looks magical and I suspect a misunderstanding somewhere.

File: ignite/metrics/gan/fid.py [hunk 1]
```diff
@@ -182,8 +182,9 @@ def reset(self) -> None:
     def update(self, output: Sequence[torch.Tensor]) -> None:
 
         train, test = output
-        train_features = self._extract_features(train)
-        test_features = self._extract_features(test)
+
+        train_features = self._extract_features(train).to(torch.float64)
+        test_features = self._extract_features(test).to(torch.float64)
```
Comment (vfdev-5) [inline ignite/metrics/gan/fid.py:187]: remove that

File: ignite/metrics/gan/fid.py [hunk 2]
```diff
@@ -182,6 +182,7 @@ def reset(self) -> None:
     def update(self, output: Sequence[torch.Tensor]) -> None:
 
         train, test = output
+
```
Comment (vfdev-5) [inline ignite/metrics/gan/fid.py:185]:
```suggestion
```

**Comment by vfdev-5** [comment]
@gucifer I'm closing this PR in favor of #2085 that should check the code agains GPU CI.

**Pull request closed without merging**